src/atoms/button/component.tsx

```diff
@@ -78,12 +78,13 @@ export interface IButtonProps extends IComponentProps<IButtonTheme> {
   onClicked?(): void;
 }
 
-export const Button = (props: IButtonProps): React.ReactElement => {
-  const onClicked = (): void => {
+export const Button = React.forwardRef((props: IButtonProps, ref: React.ForwardedRef<HTMLButtonElement>): React.ReactElement => {
```

Comment:
why is this needed?

Reply:
So that we can attach the portal to a Button

Reply:
we should only ever attach to a box - otherwise everything will need a ref at some point

Reply:
Yeah, you are right
The way to do this is to pass the prop from Button to box, right?

Reply:
yep if by "pass" you mean wrap the button in a box wherever you need it then put the ref on the box

Reply:
No, if I wrap a box on a button then how can someone click on the button.
I mean that we take the props in button and then pass it to the box component.

Reply:
That way we don't need a forward ref on button

Reply:
that doesnt make sense - if a button is in a box it can still be clicked.

Reply:
pls try and show me if it fails

Reply:
Ooh yeah, I thought of something else. If we click on a box it should propagate the click event to the parent also.
Simply a transparent box can wrap a button for this to work.

Reply:
yes exactly - box are just divs. Just FYI on this as well the box doesnt need to propogate the click explicitly - it will happen without interference from us.